### PR TITLE
Wittypixels logic

### PR DESCRIPTION
**Pull request by Tommytrg**

### CODE REVIEW

File: packages/api/src/services/canvasCache.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    //   return previousDraw
+    // })
+
+    const draws = []
+    for (let i = checkpoint; i < this.lastIndex; i += 1) {
+      const previousDraw = this.drawCache[i]
+      draws.push(previousDraw)
+    }
+
+    // TODO: Implement this logic in the frontend to be mor efficient
```
Comment (gabaldon) [inline packages/api/src/services/canvasCache.ts:47]: Should we remove this comment?

File: packages/api/src/types.ts [hunk 2]
```diff
@@ -14,16 +15,36 @@ const Palette = Type.Object({
...
   3: Type.Integer(),
   4: Type.Integer(),
+  5: Type.Integer(),
+  6: Type.Integer(),
+  7: Type.Integer(),
 })
 
+export enum Color {
+  White = 0,
+  Black = 1,
```
Comment (gabaldon) [inline packages/api/src/types.ts:25]: Is black being used?

File: packages/api/src/types.ts [hunk 3]
```diff
@@ -79,6 +127,57 @@ export const ExtendedPlayerVTO = Type.Object({
...
+    (acc, val, index) => ({
+      ...acc,
+      [index]: Type.Array(DbPixelVTO),
+    }),
+    { name: Type.Number() }
+  )
+)
+export type DbSectorVTO = Static<typeof DbSectorVTO>
+
+// export const DbCanvasVTO = Type.Array( DbSectorVTO)
```
Comment (gabaldon) [inline packages/api/src/types.ts:178]: Should we remove this comments?

File: packages/api/src/domain/draw.ts [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+      t: this.timestamp,
+      x: this.x,
+      y: this.y,
+      c: this.color,
+    }
+  }
+
+  toDbSectorInfo(): PixelLocation {
+    const x = this.x % 50
+    const y = this.y % 50
```
Comment (tmpolaczyk) [inline packages/api/src/domain/draw.ts:45]: This 50 should be `CANVAS_SECTOR_SIZE`

**Review by gabaldon — APPROVED** [review]

**Pull request merged**